pkg/handler/offchain_tip_bot_test.go

```diff
@@ -0,0 +1,244 @@
+			require.NoError(t, err)
+
+			require.JSONEq(t, string(expRespRaw), w.Body.String(), "[Handler.GetUserBalances] response mismatched")
+		})
+	}
+}
+
+func TestHandler_TransferToken(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	db := testhelper.LoadTestDB("../../migrations/test_seed")
+	repo := pg.NewRepo(db)
+	cfg := config.LoadTestConfig()
+	log := logger.NewLogrusLogger()
+	svc, _ := service.NewService(cfg, log)
+	coingeckoMock := mock_coingecko.NewMockService(ctrl)
+	svc.CoinGecko = coingeckoMock
+
+	entity := entities.New(cfg, log, repo, nil, nil, nil, nil, svc, nil, nil, nil)
+	h := &Handler{
+		entities: entity,
+		log:      log,
+	}
+
+	tests := []struct {
+		name             string
+		req              request.OffchainTransferRequest
+		wantCode         int
+		wantResponsePath string
+	}{
+		{
+			name: "tip 1 user success",
+			req: request.OffchainTransferRequest{
+				Sender:       "463379262620041226",
+				Recipients:   []string{"760874365037314100"},
+				GuildID:      "462663954813157376",
+				ChannelID:    "1003381172178530494",
+				Amount:       1.5,
+				Token:        "CAKE",
+				FullCommand:  "tip <@760874365037314100> 1.5 cake",
+				TransferType: "tip",
+			},
+			wantCode:         http.StatusOK,
+			wantResponsePath: "testdata/offchain_tip_bot/200-tip-one-user.json",
+		},
+		{
+			name: "airdrop success",
+			req: request.OffchainTransferRequest{
+				Sender:       "463379262620041226",
+				Recipients:   []string{"760874365037314100", "580788681967665173"},
+				GuildID:      "462663954813157376",
+				ChannelID:    "1003381172178530494",
+				Amount:       2,
+				Token:        "CAKE",
+				FullCommand:  "drop 2 cake",
+				TransferType: "airdrop",
+			},
+			wantCode:         http.StatusOK,
+			wantResponsePath: "testdata/offchain_tip_bot/200-airdrop.json",
+		},
+		{
+			name: "tip users success",
+			req: request.OffchainTransferRequest{
+				Sender:       "463379262620041226",
+				Recipients:   []string{"760874365037314100", "580788681967665173"},
+				GuildID:      "462663954813157376",
+				ChannelID:    "1003381172178530494",
+				Amount:       2,
+				Token:        "CAKE",
+				FullCommand:  "tip <@760874365037314100> <@580788681967665173> 2 cake",
+				TransferType: "tip",
+			},
+			wantCode:         http.StatusOK,
+			wantResponsePath: "testdata/offchain_tip_bot/200-tip-multiple-users.json",
+		},
+		{
+			name: "tip each",
+			req: request.OffchainTransferRequest{
+				Sender:       "463379262620041226",
+				Recipients:   []string{"760874365037314100", "580788681967665173"},
+				GuildID:      "462663954813157376",
+				ChannelID:    "1003381172178530494",
+				Amount:       4,
+				Each:         true,
+				Token:        "CAKE",
+				FullCommand:  "tip <@760874365037314100> <@580788681967665173> 2 cake each",
+				TransferType: "tip",
+			},
+			wantCode:         http.StatusOK,
+			wantResponsePath: "testdata/offchain_tip_bot/200-tip-each.json",
+		},
+		{
```

Comment:
can we test tip role in api, or it need be in bot ?

Reply:
role handle in bot